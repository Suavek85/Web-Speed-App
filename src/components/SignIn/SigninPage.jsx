import React, { useState } from 'react'
import { useDispatch } from "react-redux"
import classNames from 'classnames/bind'
import { useMediaQuery } from 'react-responsive'
import { navigationStates } from '../../constants/navigationStates'
import { mobileWidth } from '../../scss/mediaqueries.js'
import SigninContent from './SigninContent/SigninContent'
import SigninTabs from './SigninTabs/SigninTabs'
import styles from './SigninPage.scss'

export default function SigninPage(props) {

  const dispatch = useDispatch()
  const noAnimationOnRender = () => dispatch({ type: navigationStates.NOANIMATION })

  const [ activeTab, setActiveTab ] = useState('signin')
  const handleTabClick = e => setActiveTab(e.target.id)

  //STYLE
  const cx = classNames.bind(styles)
  
  const getClass = cx({
    block: true,
    blockMobile: useMediaQuery(mobileWidth),
  })

  return (
    <div className={ getClass }>
      <SigninTabs  activeTab={ activeTab }  handleTabClick={ handleTabClick } />
      <SigninContent 
        activeTab={activeTab} 
        handleOnClick={() => {
          noAnimationOnRender()
          props.history.push('/')
        }} 
      />
    </div>
  )
}