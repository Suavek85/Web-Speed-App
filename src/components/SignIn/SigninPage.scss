@import '../../scss/styles';

$block: main-wrapper;
$blockMobile: modifier($block, mobile);

.main-wrapper {
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  width: 400px;
  height: 660px;
  margin: 50px auto;
  background-color: var(--color-innermost-background);

  &--mobile {
    width: 100vw;
    height: auto;
    margin: 0;

    h2 {
      font-size: em(18, 16);
    }
    
  }
}

:export {
  block: $block;
  blockMobile: $blockMobile;
}
