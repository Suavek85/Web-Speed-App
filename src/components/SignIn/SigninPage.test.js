import React from 'react'
import * as redux from 'react-redux'
import { shallow } from 'enzyme'
import SigninPage from './SigninPage'

describe('SigninPage snapshot', () => {

  it('should match the snapshot', () => {
    const spyUseDispatch = jest.spyOn(redux, 'useDispatch')
    spyUseDispatch.mockReturnValue('NOANIMATION')
    const wrapper = shallow(<SigninPage />)
    expect(wrapper).toMatchSnapshot()
  })

})