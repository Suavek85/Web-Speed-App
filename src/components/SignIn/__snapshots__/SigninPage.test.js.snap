// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SigninPage snapshot should match the snapshot 1`] = `
<div
  className="block"
>
  <SigninTabs
    activeTab="signin"
    handleTabClick={[Function]}
  />
  <SigninContent
    activeTab="signin"
    handleOnClick={[Function]}
  />
</div>
`;
